CalibCalorimetry/HcalTPGAlgos/src/HcaluLUTTPGCoder.cc

```diff
@@ -595,10 +595,21 @@ void HcaluLUTTPGCoder::update(const HcalDbService& conditions) {
       const HcalLutMetadatum* meta = metadata->getValues(cell);
 
       auto tpParam = conditions.getHcalTPChannelParameter(cell, false);
-      int weight = tpParam->getauxi1();
+      const int weight = tpParam->getauxi1();
+      int factorGeVPerCount = tpParam->getauxi2();
+      if (factorGeVPerCount == 0) {
+        edm::LogWarning("HcaluLUTTPGCoder")
```

Comment:
since after this PR was merged, logs of Phase 2 step2 (`DIGI:pdigi_valid,L1TrackTrigger,L1,L1P2GT,DIGI2RAW,HLT:@relvalRun4`) workflows are full of:

```
%MSG-w HcaluLUTTPGCoder:  HcalTPGCoderULUT:HcalTPGCoderULUT@callESModule  15-Nov-2024 15:59:20 CET Run: 1 Event: 701
WARNING: ZDC trigger spacing factor, taken from auxi2 field of HCALTPChannelParameters for the cell with (zside, section, channel) =  (1 , 1 , 1) is set to the default value of 0, which is an incompatible value for a spacing factor. Setting the value to 50 and continuing.
%MSG
%MSG-w HcaluLUTTPGCoder:  HcalTPGCoderULUT:HcalTPGCoderULUT@callESModule  15-Nov-2024 15:59:20 CET Run: 1 Event: 701
WARNING: ZDC trigger spacing factor, taken from auxi2 field of HCALTPChannelParameters for the cell with (zside, section, channel) =  (1 , 1 , 2) is set to the default value of 0, which is an incompatible value for a spacing factor. Setting the value to 50 and continuing.
%MSG
%MSG-w HcaluLUTTPGCoder:  HcalTPGCoderULUT:HcalTPGCoderULUT@callESModule  15-Nov-2024 15:59:20 CET Run: 1 Event: 701
WARNING: ZDC trigger spacing factor, taken from auxi2 field of HCALTPChannelParameters for the cell with (zside, section, channel) =  (1 , 1 , 3) is set to the default value of 0, which is an incompatible value for a spacing factor. Setting the value to 50 and continuing.
%MSG
%MSG-w HcaluLUTTPGCoder:  HcalTPGCoderULUT:HcalTPGCoderULUT@callESModule  15-Nov-2024 15:59:20 CET Run: 1 Event: 701
WARNING: ZDC trigger spacing factor, taken from auxi2 field of HCALTPChannelParameters for the cell with (zside, section, channel) =  (1 , 1 , 4) is set to the default value of 0, which is an incompatible value for a spacing factor. Setting the value to 50 and continuing.
%MSG
```

see, e.g. step2 of [29634.0](https://cmssdt.cern.ch/SDT/cgi-bin/logreader/el8_amd64_gcc12/CMSSW_14_2_X_2024-11-15-1100/pyRelValMatrixLogs/run/29634.0_TTbar_14TeV+Run4D110/step2_TTbar_14TeV+Run4D110.log#/).

@perrotta @hjbossi in the PR signature there's written:

> The code will release a few lines of LogWarning message per run, until the conditions are updated with the missing content: this will be fixed at the next condition updates, and in any case LogWarning shouldn't get triggered in production-like workflows (correct?)

are we expecting this to be fixed in Phase 2 conditions as well? when is this going to happen?

Reply:
Thank you @mmusich for reporting. These logs are printed once per run: they are certainly annoying, but probably still affordable. In any case, certainly also the `HCALTPChannelParameter` tag in the Phase2 GTs should be eventually updated to take into account the second parameter searched for in those conditions.
However, my point here would be different: why ZDC is enabled at all in Phase2 workflows? Maybe the recently merged PR #46570 from @bsunanda testifies that HCAL/ZDC people is aware of the issue and they are going to provide a mechanism to solve it?

Reply:
@bsunanda do you mind letting us know the status on this? I am also not sure why ZDC is enabled in Phase 2 workflows at the moment.

Reply:
In the log of e.g. step2 of [29634.0](https://cmssdt.cern.ch/SDT/cgi-bin/logreader/el8_amd64_gcc12/CMSSW_14_2_X_2024-11-15-1100/pyRelValMatrixLogs/run/29634.0_TTbar_14TeV+Run4D110/step2_TTbar_14TeV+Run4D110.log#/), as in all other logs affected, there are warnings for the 13 ZDC channels printed once per run, as I wrote. Having the warnings printed only once per run was something that we wanted to get assured [during the review](https://github.com/cms-sw/cmssw/pull/46537#issuecomment-2449772890), in order not to flood the outputs with too much output even during this intermediate period before all relevant GTs could get updated.

This is certainly annoying, as I wrote, and it must be fixed (thank you again for having pointed it out for the Phase2 case, where it was indeed overlooked because I did not think that ZDC would have been digitized in Phase2 workflows). For "affordable" I mean that having those warnings listed only once at the begin of run should have probably still allowed to spot other warning messages, certainly in the other events of the runs, but quite likely also at that begin of run.

As examples of much less affordable warning reports let me point out (from some more or less randomly chosen workflows) the continuously repeated messages from `ElectronSeedProducer:ecalDrivenElectronSeeds` in [step3 of wf 136.874](https://cmssdt.cern.ch/SDT/cgi-bin/logreader/el8_amd64_gcc12/CMSSW_14_2_X_2024-11-13-1300/pyRelValMatrixLogs/run/136.874_RunEGamma2018C/step3_RunEGamma2018C.log#/), or the ones from `TrackRefitter` or `PrimaryVertexAnalyzer4PUSlimmed` (and probably also a few more) from [step3 of wf 181.1](https://cmssdt.cern.ch/SDT/cgi-bin/logreader/el8_amd64_gcc12/CMSSW_14_2_X_2024-11-13-1300/pyRelValMatrixLogs/run/181.1_Starlight_DoubleDiffraction_5360_HI_2024/step3_Starlight_DoubleDiffraction_5360_HI_2024.log#/329), which are there from long time and apparently nobody cares about.